mods/cnc/maps/gdi05c/gdi05c.lua

```diff
@@ -0,0 +1,154 @@
+	{ types = { ltnk = 1 }      , route = AtkRoute3 },
+	{ types = { bggy = 1 }      , route = AtkRoute4 }
+}
+
+Atk1Delay = { easy = DateTime.Seconds(60), normal = DateTime.Seconds(50), hard = DateTime.Seconds(40) }
+Atk2Delay = { easy = DateTime.Seconds(90), normal = DateTime.Seconds(70), hard = DateTime.Seconds(50) }
+Atk3Delay = { easy = DateTime.Seconds(120), normal = DateTime.Seconds(90), hard = DateTime.Seconds(70) }
+Atk4Delay = { easy = DateTime.Seconds(150), normal = DateTime.Seconds(110), hard = DateTime.Seconds(90) }
+AutoAtkStartDelay = { easy = DateTime.Seconds(150), normal = DateTime.Seconds(120), hard = DateTime.Seconds(90) }
+AutoAtkMinDelay = { easy = DateTime.Minutes(1), normal = DateTime.Seconds(45), hard = DateTime.Seconds(30) }
+AutoAtkMaxDelay = { easy = DateTime.Minutes(2), normal = DateTime.Seconds(90), hard = DateTime.Minutes(1) }
+
+GDIBase = { GdiNuke1, GdiProc1, GdiWeap1, GdiNuke2, GdiNuke3, GdiPyle1, GdiSilo1, GdiRadar1 }
+GDIUnits = { "e2", "e2", "e2", "e2", "e1", "e1", "e1", "e1", "mtnk", "mtnk", "jeep", "apc", "apc" }
+GDIHarvester = { "harv" }
+NodSams = { Sam1, Sam2, Sam3 }
+
+AutoCreateTeam = function()
+	local team = Utils.Random(AutoCreateTeams)
+	for type, count in pairs(team.types) do
+		MoveAndHunt(Utils.Take(count, Nod.GetActorsByType(type)), team.route)
+	end
+
+	Trigger.AfterDelay(Utils.RandomInteger(AutoAtkMinDelay[Difficulty], AutoAtkMaxDelay[Difficulty]), AutoCreateTeam)
+end
+
+DiscoverGDIBase = function(actor, discoverer)
+	if BaseDiscovered or not discoverer == GDI then
+		return
+	end
+
+	--Spawn harv only when base is discovered, otherwise it will waste tiberium and might get killed before the player arrives
+	Reinforcements.Reinforce(GDI, GDIHarvester, { GdiProc1.Location + CVec.New(0, 2), GdiProc1.Location + CVec.New(-1, 3) }, 1)
+
+	Utils.Do(GDIBase, function(actor)
+		actor.Owner = GDI
+	end)
+
+	BaseDiscovered = true
+
+	EliminateNod = GDI.AddObjective("Eliminate all Nod forces in the area.")
+	GDI.MarkCompletedObjective(FindBase)
+end
+
+Atk1TriggerFunction = function()
+	MoveAndHunt(Utils.Take(2, Nod.GetActorsByType('e1')), AtkRoute1)
+	MoveAndHunt(Utils.Take(2, Nod.GetActorsByType('e3')), AtkRoute1)
+end
+
+Atk2TriggerFunction = function()
+	MoveAndHunt(Utils.Take(4, Nod.GetActorsByType('e3')), AtkRoute2)
+end
+
+Atk3TriggerFunction = function()
+	MoveAndHunt(Utils.Take(1, Nod.GetActorsByType('bggy')), AtkRoute2)
+end
+
+Atk4TriggerFunction = function()
+	MoveAndHunt(Utils.Take(2, Nod.GetActorsByType('e1')), AtkRoute1)
+	MoveAndHunt(Utils.Take(1, Nod.GetActorsByType('ltnk')), AtkRoute1)
+end
+
+InsertGDIUnits = function()
+	Media.PlaySpeechNotification(GDI, "Reinforce")
+	Reinforcements.Reinforce(GDI, GDIUnits, { UnitsEntry.Location, UnitsRally.Location }, 15)
+end
+
+WorldLoaded = function()
+	GDI = Player.GetPlayer("GDI")
+	AbandonedBase = Player.GetPlayer("AbandonedBase")
+	Nod = Player.GetPlayer("Nod")
+
+	InitObjectives(GDI)
+
+	RepairNamedActors(Nod, RepairThreshold[Difficulty])
```

Comment:
Oh, and you're missing a `Difficulty = Map.LobbyOption("difficulty")`. Interestingly, this doesn't cause a crash on your branch, but does when it gets rebased onto bleed.

Reply:
Yes, because that line was removed from the global script...